whats_fresh/whats_fresh_api/tests/models/test_theme_model.py

```diff
@@ -0,0 +1,154 @@
+            'font_color': models.TextField,
+            'logo': models.ImageField,
+            'slogan': models.CharField,
+            'site_title': models.CharField,
+            'vendors': models.CharField,
+            'products': models.CharField,
+            'preparations': models.CharField,
+            'stories': models.CharField,
+            'images': models.CharField,
+            'videos': models.CharField,
+            'vendors_slug': models.SlugField,
+            'products_slug': models.SlugField,
+            'preparations_slug': models.SlugField,
+            'stories_slug': models.SlugField,
+            'images_slug': models.SlugField,
+            'videos_slug': models.SlugField,
+            'active': models.CharField,
+            'id': models.AutoField
+        }
+
+        self.optional_fields = {
+            'logo',
+            'slogan'
+        }
+
+        self.null_fields = {'logo', 'slogan'}
+
+    def test_fields_exist(self):
+        model = models.get_model('whats_fresh_api', 'Theme')
+        for field, field_type in self.expected_fields.items():
+            self.assertEqual(
+                field_type, type(model._meta.get_field_by_name(field)[0]))
+
+    def test_no_additional_fields(self):
+        fields = Theme._meta.get_all_field_names()
+        self.assertTrue(sorted(fields) == sorted(self.expected_fields.keys()))
+
+    def test___unicode___method(self):
+        try:
+            Theme.__unicode__(Theme())
+        except AttributeError:
+            self.fail("No __unicode__ method found")
+
+    def test_optional_fields(self):
+        models.get_model('whats_fresh_api', 'Theme')
+        for field in self.optional_fields:
+            self.assertEqual(
+                Theme._meta.get_field_by_name(field)[0].blank, True)
+        for field in self.null_fields:
+            self.assertEqual(
+                Theme._meta.get_field_by_name(field)[0].null, True)
+
+    def test_slug_changes(self):
+
+        form = ThemeAdminForm(data={
+            'name': 'Test Name',
+            'background_color': 'rgb(81, 114, 133)',
+            'foreground_color': 'rgb(81, 114, 133)',
+            'header_color': 'rgb(255, 255, 255)',
+            'font_color': 'rgb(51, 51, 51)',
+            'logo': 'cat.jpg',
+            'slogan': 'Test Slogan',
+            'site_title': 'Test Title',
+            'vendors': 'vendors',
+            'products': 'products',
+            'preparations': 'preparations',
+            'stories': 'stories',
+            'images': 'images',
+            'videos': 'videos',
+            'vendors_slug': 'test-vendors',
+            'products_slug': 'test-products',
+            'preparations_slug': 'test-preparations',
+            'stories_slug': 'test-stories',
+            'images_slug': 'test-images',
+            'videos_slug': 'test-videos',
+            'active': 'No'})
+
+        form.save()
+
+        self.assertEqual(reverse('vendors-list'), '/1/test-vendors')
```

Comment:
@tschuy We should get rid of this method, right? As it would always never pass because server needs to be restarted whenever slug values are changed?